cmd.cpp

```diff
@@ -1,39 +1,44 @@
 #include "cmd.h"
+#include <iostream>
+#include <fstream>
 using namespace std;
 
 void Dictionary::mkdir(string dirName, string currentFolder) {
 	bool found = false;
-	for (unsigned int i = 0; i < this->system.size(); i++) {
-		if (currentFolder == this->system[i].folder && dirName == this->system[i].subfolder) {
-			found = true;
+	if (this->checkIfNameIsValid(dirName)) cout << "Name cannot contain special characters" << endl;
+	else {
+		for (unsigned int i = 0; i < this->system.size(); i++) {
+			if (currentFolder == this->system[i].folder && dirName == this->system[i].subfolder) {
+				found = true;
+			}
 		}
-	}
-	for (unsigned int i = 0; i < this->fileDescriptorVector.size(); i++) {
-		if (dirName == this->fileDescriptorVector[i].fileName && currentFolder == this->fileDescriptorVector[i].filePath) {
-			found = true;
+		for (unsigned int i = 0; i < this->fileDescriptorVector.size(); i++) {
+			if (dirName == this->fileDescriptorVector[i].fileName && currentFolder == this->fileDescriptorVector[i].filePath) {
+				found = true;
+			}
 		}
-	}
-	if (found) {
-		cout << "This file/directory already exists" << endl;
-	}
-	else
-	{
-		if (dirName != "..") {
-			Pair p;
-			p.folder = currentFolder;
-			p.subfolder = dirName;
-			this->system.push_back(p);
+		if (found) {
+			cout << "This file/directory already exists" << endl;
 		}
-		else {
-			cout << "Invalid foldername" << endl;
+		else
+		{
+			if (dirName != "..") {
+				this->addPairToVector(currentFolder, dirName);
+			}
+			else {
+				cout << "Invalid foldername" << endl;
```

Comment:
szoval a `..` az egyetlen invalid foldername?

Reply:
True, ellenőrzö függvényt raktam ide, ami ellenőrzi. Az elnevezése nem volt a legmegfelelőbb, azt is javítottam.